lib/providers/event_details/team_traces_notifier.dart

```diff
@@ -0,0 +1,53 @@
+import 'package:firebase_database/firebase_database.dart';
+import 'package:regatta_buddy/providers/firebase_providers.dart';
+import 'package:regatta_buddy/utils/logging/logger_helper.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+import 'package:latlong2/latlong.dart';
+
+part 'team_traces_notifier.g.dart';
+
+@Riverpod(keepAlive: true)
+class TeamTracesNotifier extends _$TeamTracesNotifier {
+
+  @override
+  Map<String, Map<int, List<LatLng>>> build(String eventId) {
+    final firebaseDb = ref.watch(firebaseDbProvider);
+    final logger = getLogger("TeamTracesNotifier");
+    logger.i("initializing a TeamTracesNotifier provider");
+
+    final dbScoresRef = firebaseDb.child('scores').child(eventId);
+
+    dbScoresRef.onValue.listen((event) {
+      final teamsAndScores = event.snapshot.value as Map<dynamic, dynamic>;
+
+      for (final round in List.generate(10, (index) => index)) {
+        for (final team in teamsAndScores.keys.toList()) {
+          final dbRef = firebaseDb
+              .child('traces')
+              .child(eventId)
+              .child(team)
+              .child('positions')
+              .child('rounds')
+              .child(round.toString());
```

Comment:
Right now we are starting to have a lot of child defined by strings. Maybe we should start thinking on creating functions that in this example would get eventId, team, and round and return reference to us. What do you think?

Reply:
Honestly, I say that it would be unnecessary. The current usage of multiple "child()"s is readable and easily understandable